### PR TITLE
Feature/peter/is job complete

### PR DESCRIPTION
**Pull request by reteppeter**
# 
## Proposed changes

Adds a new service, isJobComplete.
This service is run every time OwlEyes or Seenomaly processes an image/video of a batch job. 

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] DevOps related

## Checklist

- [x] I have read the [project guidelines](https://github.com/Visual-Testing-for-Android-Apps/Project-guidelines) doc
- [x] I locally linted my code
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

## Further comments

There are issues with our current linting setup:
- ```Strings must use singlequote.```. The default value is "double", but I don't think you changed it anywhere.
- It complains about semicolons, but our formatter, prettier, is what added them.
- The cloud formation error is to do with Seenomaly.

### CODE REVIEW

**Comment by reteppeter** [comment]
I don't understand the linting errors at all.
- ```Strings must use singlequote.```. The default value is "double", but I don't think you changed it anywhere.
- It complains about semicolons, but the linting (on my machine) is what added them.
- The cloud formation error is to do with Seenomaly.

File: server/isJobComplete/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+			};
+			await updateJob(request);
+
+			//const message: String = '{ "jobKey": "' + String(key) + '" }'
+
+			//Push a request to the SQS queue
+			const params: SendMessageRequest = {
+				MessageBody: '{ "jobKey": "' + String(key) + '" }',
+				QueueUrl: process.env.REPORT_GENERATION_QUEUE as string,
+			};
```
Comment (alexdelicato) [inline server/isJobComplete/src/index.ts:95]: In future where we want to send an email letting the user know their job is processing, can it be done here, similar to the lines above but for SEND_EMAIL_QUEUE ?

File: server/isJobComplete/src/service/dynamodbService.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+//           id: uuidv4,
+//           email,
+//           videoFiles,
+//           imageFiles,
+//           updatedAt: new Date().toISOString()
+//       }
+//     } as PutItemInput
+
+//     await putItem(newJobItem)
+//   }
```
Comment (alexdelicato) [inline server/isJobComplete/src/service/dynamodbService.ts:24]: should this file be deleted if everything is commented out?

File: server/template.yml [hunk 3]
```diff
@@ -181,42 +186,42 @@ Resources:
...
+          Properties:
+            Path: /fileCompress
+            Method: get
 
   SendEmail:
     Type: AWS::Serverless::Function
     Properties:
       CodeUri: sendEmail/
-      Handler: dist/send-email.sendEmail
+      Handler: dist/index.sendEmail
```
Comment (alexdelicato) [inline server/template.yml:224]: is this a standard to have the name of file as index?

File: server/template.yml [hunk 4]
```diff
@@ -294,8 +320,8 @@ Outputs:
     Value: !Sub "https://${ServerlessRestApi}.execute-api.${AWS::Region}.amazonaws.com/prod/"
   UploadHandlerFunction:
     Value: !Ref UploadHandler
-  PostProcessHandlerFuntion:
-    Value: !Ref PostProcessHandler
+  JobCompletionStatusFuntion:
+    Value: !Ref JobCompletionStatus
   FileCompressApi:
     Description: "API Gateway endpoint URL for Prod stage for file compression function"
     Value: !Sub "https://${ServerlessRestApi}.execute-api.${AWS::Region}.amazonaws.com/Prod/fileCompress/"
```
Comment (alexdelicato) [inline server/template.yml:327]: I'm not exactly sure why, but the send email one is missing which is in develop branch
```
 SendEmailFunction:
    Value: !Ref SendEmail
```

**Review by alexdelicato — CHANGES_REQUESTED** [review]

**Review by beining00 — APPROVED** [review]

File: server/isJobComplete/src/index.ts (hunk 1 above)
Comment (reteppeter) [reply server/isJobComplete/src/index.ts:95]: Yes, probably by just calling your function with a different body as well.

File: server/template.yml (hunk 3 above)
Comment (reteppeter) [reply server/template.yml:224]: Yep. I renamed your file, so it should be in the same place but called index.ts.

File: server/template.yml (hunk 4 above)
Comment (reteppeter) [reply server/template.yml:327]: It's present identically in both.

**Review by wil-vanderstaay — APPROVED** [review]

**Review by alexdelicato — APPROVED** [review]

**Pull request merged**